### PR TITLE
Create single instance of empty BarCode + naming cleanup

### PR DESCRIPTION
**Pull request by pavlospt**
Moved empty Strings to private static final variables

### CODE REVIEW

File: store/src/main/java/com/nytimes/android/external/store/base/impl/BarCode.java [hunk 1]
```diff
@@ -24,7 +28,7 @@ public BarCode(@Nonnull String type, @Nonnull String key) {
 
     @Nonnull
     public static BarCode empty() {
-        return new BarCode("", "");
```
Comment (PaulWoitaschek) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/BarCode.java]: BarCode is immutable, better create a static instancd of the empty barcode and return that.

**Comment by pavlospt** [comment]
@digitalbuddha Yeah I got the project open now and just fixing things, expect some more for today 😄

**Comment by digitalbuddha** [comment]
I plan to cut 2.0 early next week. Lmk if any name/api change are needed
before we go live.

File: store/src/main/java/com/nytimes/android/external/store/base/impl/BarCode.java (hunk 1 above)
Comment (pavlospt) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/BarCode.java]: @PaulWoitaschek  Thanks! Done!

**Comment by pavlospt** [comment]
@digitalbuddha Yeah will be on time till then! No worries!

**Comment by digitalbuddha** [comment]
I don't think we need the other static strings. Especially if they are the same.

**Comment by pavlospt** [comment]
@digitalbuddha I did in case an empty type and empty key have different definitions in the future!

**Comment by PaulWoitaschek** [comment]
Thats just an implementation detail and does not make the code more readable or maintainable. In that case the shorter version is better.

Especially now that you eliminated the unnecessary object creation by making a single instance of the empty barcode.

**Comment by pavlospt** [comment]
@PaulWoitaschek Ok great then!

**Pull request merged**